src/bin/rav1e-ch.rs

```diff
@@ -0,0 +1,574 @@
+    }
+  }
+
+  // this can be changed to flatten
+  let level = std::env::var("RAV1E_LOG")
+    .ok()
+    .map(|l| log::LevelFilter::from_str(&l).ok())
+    .unwrap_or(Some(log::LevelFilter::Info))
+    .unwrap();
+
+  fern::Dispatch::new()
+    .format(move |out, message, record| {
+      out.finish(format_args!(
+        "{level} {message}",
+        level = level_colored(record.level()),
+        message = message,
+      ));
+    })
+    // set the default log level. to filter out verbose log messages from dependencies, set
+    // this to Warn and overwrite the log level for your crate.
+    .level(log::LevelFilter::Warn)
+    // change log levels for individual modules. Note: This looks for the record's target
+    // field which defaults to the module path but can be overwritten with the `target`
+    // parameter:
+    // `info!(target="special_target", "This log message is about special_target");`
+    .level_for("rav1e", level)
+    // output to stdout
+    .chain(std::io::stderr())
+    .apply()
+    .unwrap();
+}
+
+cfg_if::cfg_if! {
+  if #[cfg(any(target_os = "windows", target_arch = "wasm32"))] {
+    fn print_rusage() {
+      eprintln!("Windows benchmarking is not supported currently.");
+    }
+  } else {
+    fn print_rusage() {
+      let (utime, stime, maxrss) = unsafe {
+        let mut usage = std::mem::zeroed();
+        let _ = libc::getrusage(libc::RUSAGE_SELF, &mut usage);
+        (usage.ru_utime, usage.ru_stime, usage.ru_maxrss)
+      };
+      eprintln!(
+        "user time: {} s",
+        utime.tv_sec as f64 + utime.tv_usec as f64 / 1_000_000f64
+      );
+      eprintln!(
+        "system time: {} s",
+        stime.tv_sec as f64 + stime.tv_usec as f64 / 1_000_000f64
+      );
+      eprintln!("maximum rss: {} KB", maxrss);
+    }
+  }
+}
+
+fn run() -> Result<(), error::CliError> {
+  let mut cli = parse_cli()?;
+  // Maximum frame size by specification + maximum y4m header
+  let limit = y4m::Limits {
+    // Use saturating operations to gracefully handle 32-bit architectures
+    bytes: 64usize
+      .saturating_mul(64)
+      .saturating_mul(4096)
+      .saturating_mul(2304)
+      .saturating_add(1024),
+  };
+  let mut y4m_dec = match y4m::Decoder::new_with_limits(cli.io.input, limit) {
+    Err(_) => {
+      return Err(CliError::new("Could not input video. Is it a y4m file?"))
+    }
+    Ok(d) => d,
+  };
+  let video_info = y4m_dec.get_video_details();
+  let y4m_enc = match cli.io.rec {
+    Some(rec) => Some(
+      y4m::encode(
+        video_info.width,
+        video_info.height,
+        y4m::Ratio::new(
+          video_info.time_base.den as usize,
+          video_info.time_base.num as usize,
+        ),
+      )
+      .with_colorspace(y4m_dec.get_colorspace())
+      .write_header(rec)
+      .unwrap(),
+    ),
+    None => None,
+  };
+
+  match video_info.bit_depth {
+    8 | 10 | 12 => {}
+    _ => return Err(CliError::new("Unsupported bit depth")),
```

Comment:
Are there other bit depths? :exploding_head:

Reply:
Theoretically input could be 16- or 32-bit. But rav1e doesn't support it. (I think even AV1 doesn't support it? Haven't checked.)

Reply:
Yep